app/controllers/reservations_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ReservationsController < ApplicationController
+  before_action :logged_in_user, only: %i[index show]
   def index
-    @reserved_books = Reservation.where(user_id: current_user.id).where("reservation_at > ?",
+    @reservations = Reservation.where(user_id: current_user.id).where("reservation_at >= ?",
                                                                         Time.now).order(reservation_at: :asc)
   end
 
   def show
+    @reservation = Reservation.find(params[:id])
+    return unless @reservation.nil?
+
+    redirect_to book_path(@reservation.book.id)
```

Comment:
```suggestion
    redirect_to book_path(@reservation.book) unless @reservation
```
にまとめてもいいかもです 🤔 

読みにくければ、`@reservation.book.id`の`id`だけ消してもらえると 🙏